common/hashing.h

```diff
@@ -134,26 +135,23 @@ class HashCode : public Printable<HashCode> {
 // - `llvm::SmallString`
 //
 // This function supports heterogeneous lookup between all of the string-like
-// types. However, this is the only heterogeneous lookup support including for
-// the builtin in, standard, and LLVM types. Notably, each different size and
+// types. It also supports heterogeneous lookup between pointer types regardless
+// of pointee type and `nullptr`.
+//
+// However, these are the only heterogeneous lookup support including for the
+// builtin in, standard, and LLVM types. Notably, each different size and
 // signedness integer type may hash differently for efficiency reasons. Hash
 // tables should pick a single integer type in which to manage keys and do
 // lookups.
 //
 // To add support for your type, you need to implement a customization point --
-// a free function that can be found by ADL for your type -- called `CarbonHash`
-// with the following signature:
+// a free function that can be found by ADL for your type -- called
+// `CarbonHashValue` with the following signature:
 //
 // ```cpp
-// auto CarbonHash(const YourType& value, uint64_t seed) -> HashCode;
+// auto CarbonHashValue(const YourType& value, uint64_t seed) -> HashCode;
```

Comment:
There should at least be a breadcrumb here pointing to the contract that this function needs to satisfy, especially now that the signature doesn't mention `Hasher`. However, I think it would be clearer to have the full contract here rather than on `Hasher`, possibly with something at the end like "`CarbonHashValue` overloads should normally be implemented using the `Hasher` utility class below."